fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -384,12 +391,16 @@ class BrowserRobot {
 
     fun clickSelectAddressButton() {
         for (i in 1..RETRY_COUNT) {
+            Log.i(TAG, "clickSelectAddressButton: For loop i = $i")
             try {
+                Log.i(TAG, "clickSelectAddressButton: Try block")
```

Comment:
You can merge these 2 logs into something like "Started try #$i". Then inside the catch block, you can say "Assertion failed caught, executing fallback methods"

Reply:
Done !